src/Summoner/Template.hs

```diff
@@ -381,53 +383,83 @@ createStackTemplate ProjectData{..} = Dir (toString repo) $
         cache:
           directories:
           - "$$HOME/.stack"
-          - "$$HOME/build/${owner}/${repo}/.stack-work"
+          - "$$TRAVIS_BUILD_DIR/.stack-work"
 
         matrix:
           include:
 
-          $travisMtr
+          $travisCabalMtr
+          $travisStackMtr
 
           - ghc: $defGhc
-            env: GHCVER='${defGhc}' STACK_YAML="$$HOME/build/${owner}/${repo}/stack.yaml"
-
-        addons:
-          apt:
-            sources:
-              - sourceline: 'ppa:hvr/ghc'
-            packages:
-              - libgmp-dev
-
-        before_install:
-          - mkdir -p ~/.local/bin
-          - export PATH="$$HOME/.local/bin:$$PATH"
-          - travis_retry curl -L 'https://www.stackage.org/stack/linux-x86_64' | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
-          - stack --version
-
+            env: GHCVER='${defGhc}' STACK_YAML="$$TRAVIS_BUILD_DIR/stack.yaml"
+            os: linux
+            addons:
+              apt:
+                packages:
+                - libgmp-dev
 
         install:
-          - travis_wait 30 stack setup --no-terminal
-          - stack ghc -- --version
-
-          - travis_wait 40 stack build --only-dependencies --no-terminal
-          - travis_wait 40 stack build --test --bench --haddock --no-run-tests --no-run-benchmarks --no-haddock-deps --no-terminal
-
+          - |
+            if [ -z "$$STACK_YAML" ]; then
+              export PATH="/opt/ghc/$$GHCVER/bin:/opt/cabal/$$CABALVER/bin:$$PATH"
+              echo $$PATH
+              cabal new-update
+              cabal new-build --enable-tests --enable-benchmarks
+            else
+              mkdir -p ~/.local/bin
+              export PATH="$$HOME/.local/bin:$$PATH"
+              travis_retry curl -L 'https://www.stackage.org/stack/linux-x86_64' | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
+              stack --version
+              stack setup --no-terminal --install-cabal 2.0.1.0
+              stack ghc -- --version
+              stack build --only-dependencies --no-terminal
+            fi
         script:
-          - travis_wait 40 stack build --test --no-terminal
+          - |
+            if [ -z "$$STACK_YAML" ]; then
+               ${cabalTest}
+            else
+              stack build --test --bench --no-run-benchmarks --no-terminal
+            fi
 
         notifications:
           email: false
         $endLine
         |]
 
-    travisMatrixItem :: Text -> Text
-    travisMatrixItem ghcV =
+    cabalTest :: Text
+    cabalTest = if test then "cabal new-test" else "echo 'No tests'"
```

Comment:
Looks like it deserves issue in `cabal` (if it doesn't exist yet) :pepe:

Reply:
https://github.com/haskell/cabal/issues/5435